github/repos_contents.go

```diff
@@ -267,7 +267,11 @@ func (s *RepositoriesService) GetArchiveLink(owner, repo string, archiveformat a
 	} else {
 		resp, err = s.client.client.Transport.RoundTrip(req)
 	}
-	if err != nil || resp.StatusCode != http.StatusFound {
+	if err != nil {
+		return nil, nil, err
+	}
+	resp.Body.Close()
+	if resp.StatusCode != http.StatusFound {
 		return nil, newResponse(resp), err
```

Comment:
The value of `err` will always be `nil` here, so you're doing `return nil, newResponse(resp), nil` effectively.

We want to return a non-nil error here, since this is an error condition and there's no valid url being returned.